File: backend/app/dtos/optimization_read.py
from pydantic import BaseModel
from .character_plan_item import CharacterPlanItem
from .character_optimization_read import CharacterOptimizationRead

class OptimizationRead(BaseModel):
    expansion_id: int
    total_profit: int
    crafts: list[CharacterPlanItem]
    characters: list[CharacterOptimizationRead]

File: frontend/src/dto/OptimizationRead.ts
import type { CharacterOptimizationRead } from "./CharacterOptimizationRead";
import type { CharacterPlanItem } from "./CharacterPlanItem";

export interface OptimizationRead {
    expansion_id: number;
    total_profit: number;
    crafts: CharacterPlanItem[];
    characters: CharacterOptimizationRead[];
}
